core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -275,4 +483,12 @@ case class P2SHNestedSegwitV0UTXOSpendingInfo(
   override val redeemScriptOpt: Option[ScriptPubKey] = Some(redeemScript)
 
   override val scriptWitnessOpt: Option[ScriptWitnessV0] = Some(scriptWitness)
+
+  override def nestedSpendingInfo: SegwitV0NativeUTXOSpendingInfo =
```

Comment:
nit: can be a `lazy val` or `val`

Reply:
I've actually been wrestling with this a lot, I'm starting to think that these should be `val`s so that you get an error right away if anything is wrong, rather than `lazy` or `def` which is perhaps more performant, thoughts?